SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -788,9 +413,38 @@ CEulerSolver::CEulerSolver(CGeometry *geometry, CConfig *config, unsigned short
 
   /*--- Initialize the solution to the far-field state everywhere. ---*/
 
-  nodes = new CEulerVariable(Density_Inf, Velocity_Inf, Energy_Inf, nPoint, nDim, nVar, config);
+  if (navier_stokes) {
+    nodes = new CNSVariable(Density_Inf, Velocity_Inf, Energy_Inf, nPoint, nDim, nVar, config);
+  } else {
+    nodes = new CEulerVariable(Density_Inf, Velocity_Inf, Energy_Inf, nPoint, nDim, nVar, config);
+  }
   SetBaseClassPointerToNodes();
 
+#ifdef HAVE_OMP
+  /*--- Get the edge coloring, on coarse grids (which are difficult to color) setup the reducer strategy,
+   *    if required (i.e. in parallel). One loop is performed over edges followed by a point loop to sum
+   *    the fluxes for each cell and set the diagonal of the system matrix. ---*/
+
+  const auto& coloring = geometry->GetEdgeColoring();
+
+  if (!coloring.empty()) {
+    auto nColor = coloring.getOuterSize();
+    EdgeColoring.resize(nColor);
+
+    for(auto iColor = 0ul; iColor < nColor; ++iColor) {
+      EdgeColoring[iColor].size = coloring.getNumNonZeros(iColor);
+      EdgeColoring[iColor].indices = coloring.innerIdx(iColor);
+    }
+  }
+  ColorGroupSize = geometry->GetEdgeColorGroupSize();
+
+  if ((MGLevel != MESH_0) && (omp_get_max_threads() > 1)) {
+    EdgeFluxes.Initialize(geometry->GetnEdge(), geometry->GetnEdge(), nVar, nullptr);
+  }
+
+  omp_chunk_size = computeStaticChunkSize(nPoint, omp_get_max_threads(), OMP_MAX_SIZE);
+#endif
```

Comment:
When in parallel, coarse grids initialise this CSysVector to store edge fluxes, the coloring they obtain from CGeometry is the natural color, i.e. a single color with all edges which is equivalent to `for(iEdge=0; iEdge < nEdge; ...)`